File: src/main/java/cz/vhromada/catalog/gui/common/Picture.java
package cz.vhromada.catalog.gui.common;

import javax.swing.ImageIcon;

/**
 * An enumeration represents picture.
 *
 * @author Vladimir Hromada
 */
public enum Picture {

    /**
     * About picture
     */
    ABOUT(new ImageIcon("pics/about.jpg")),

    /**
     * Add picture
     */
    ADD(new ImageIcon("pics/add.jpg")),

    /**
     * Cancel picture
     */
    CANCEL(new ImageIcon("pics/cancel.jpg")),

    /**
     * Catalog application picture
     */
    CATALOG(new ImageIcon("pics/catalog.jpg")),

    /**
     * Choose picture
     */
    CHOOSE(new ImageIcon("pics/choose.jpg")),

    /**
     * Move down picture
     */
    DOWN(new ImageIcon("pics/down.jpg")),

    /**
     * Duplicate picture
     */
    DUPLICATE(new ImageIcon("pics/duplicate.jpg")),

    /**
     * Exit application picture
     */
    EXIT(new ImageIcon("pics/exit.jpg")),

    /**
     * New data picture
     */
    NEW(new ImageIcon("pics/new.jpg")),

    /**
     * OK picture
     */
    OK(new ImageIcon("pics/ok.jpg")),

    /**
     * Remove
     */
    REMOVE(new ImageIcon("pics/remove.jpg")),

    /**
     * Save picture
     */
    SAVE(new ImageIcon("pics/save.jpg")),

    /**
     * Move up picture
     */
    UP(new ImageIcon("pics/up.jpg")),

    /**
     * Update picture
     */
    UPDATE(new ImageIcon("pics/update.jpg"));

    /**
     * Icon
     */
    private final ImageIcon icon;

    /**
     * Creates a new instance of Pictures.
     *
     * @param icon icon
     */
    Picture(final ImageIcon icon) {
        this.icon = icon;
    }

    /**
     * Returns icon.
     *
     * @return icon
     */
    public ImageIcon getIcon() {
        return icon;
    }

}

File: src/main/kotlin/cz/vhromada/catalog/gui/common/Picture.kt
package cz.vhromada.catalog.gui.common

import javax.swing.ImageIcon

/**
 * An enumeration represents picture.
 *
 * @author Vladimir Hromada
 */
enum class Picture(val icon: ImageIcon) {

    /**
     * About picture
     */
    ABOUT(ImageIcon("pics/about.jpg")),

    /**
     * Add picture
     */
    ADD(ImageIcon("pics/add.jpg")),

    /**
     * Cancel picture
     */
    CANCEL(ImageIcon("pics/cancel.jpg")),

    /**
     * Catalog application picture
     */
    CATALOG(ImageIcon("pics/catalog.jpg")),

    /**
     * Choose picture
     */
    CHOOSE(ImageIcon("pics/choose.jpg")),

    /**
     * Move down picture
     */
    DOWN(ImageIcon("pics/down.jpg")),

    /**
     * Duplicate picture
     */
    DUPLICATE(ImageIcon("pics/duplicate.jpg")),

    /**
     * Exit application picture
     */
    EXIT(ImageIcon("pics/exit.jpg")),

    /**
     * New data picture
     */
    NEW(ImageIcon("pics/new.jpg")),

    /**
     * OK picture
     */
    OK(ImageIcon("pics/ok.jpg")),

    /**
     * Remove
     */
    REMOVE(ImageIcon("pics/remove.jpg")),

    /**
     * Save picture
     */
    SAVE(ImageIcon("pics/save.jpg")),

    /**
     * Move up picture
     */
    UP(ImageIcon("pics/up.jpg")),

    /**
     * Update picture
     */
    UPDATE(ImageIcon("pics/update.jpg"))

}
